### PR TITLE
Uniform cursor unfocused rendering. #4441

### PR DESCRIPTION
**Pull request by tisilent**
#4441

### CODE REVIEW

**Review by Tyriar — COMMENTED** [review]
I remembered when looking at this that we actually did the opposite a while ago, tracking it down  I see it was done 3.5 years ago:

- https://github.com/xtermjs/xterm.js/pull/2455
- https://github.com/xtermjs/xterm.js/issues/2356
- https://github.com/microsoft/vscode/issues/67410

This is something that the expected behavior isn't well defined. Personally I think it's better with the rectangle, regardless of cursor style so it's obvious that it's not focused. @jerch what do you think? Should we merge this and close any future complaints as by design?

**Comment by tisilent** [comment]
I was just thinking about whether to comment or delete the code. :flushed:

**Comment by jerch** [comment]
Hmm hard to tell, whats the right direction here. Imho there are two aspects that a hollow cursor will show - the current cursor pos + the fact that the terminal in not focused atm.

I cannot think of a good pendant for underline/bar cursors here - still showing the underline/bar loses the unfocused state information for the user. And while not showing any cursor would transport that info, it now lost the position info for the user. So imho the hollow rectangle is the best compromise, since it can show the user both aspects.

I also think that ppl would not mind much, if a terminal they are not actively interacting with would flip the cursor into a hollow block? (Ofc the cursor should not magically appear, if requested to be hidden before by a sequence.)

To sum things up - I think the hollow rectangle is the right way to go for unfocused state.

**Review by Tyriar — APPROVED** [review]

**Comment by boettges** [comment]
> This is something that the expected behavior isn't well defined. Personally I think it's better with the rectangle, regardless of cursor style so it's obvious that it's not focused. 

@tyriar also regarding: https://github.com/microsoft/vscode/issues/178379#issuecomment-1487169731 

Since applying custom CSS styles is not possible for terminal's content, I think there should be a choice.

Instead of forcing "block" for all inactive terminal instances, I'd suggest a VS Code user setting (for example `terminal.integrated.inactiveCursorStyle`) with the following options:

- `outline` (default)
- `line` (non-blinking)
- `underline` (non-blinking)
- `none` (cursor hidden)

Personally, I've gotten used to the blinking line in regular input modes, while I use both, the block and outline cursor, exclusively in vim's normal and visual mode, respectively.

**Comment by Tyriar** [comment]
@boettges sounds good, created https://github.com/xtermjs/xterm.js/issues/4566
